plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesbuilder/N4JSClassDeclarationTypesBuilder.xtend

```diff
@@ -94,7 +93,7 @@ public class N4JSClassDeclarationTypesBuilder extends N4JSClassifierDeclarationT
 
 		n4Class.definedType = tclass;
 
-		target.internalTypes += tclass;
+		target.containingModule.internalTypes += tclass;
```

Comment:
I find this confusing. It seems `target` is sometimes the target (i.e. the `AbstractNamespace` that will contain the new class) and sometimes the sibling of the new class.

Reply:
What would be an example of `target` being a sibling of a new class here?

Reply:
Sorry, I should've explained better what I meant: `target.containingModule` sometimes returns `target` itself, and sometimes the parent of `target`. Depending on these two cases, the role of an argument passed in for parameter `target` changes: in the former case the newly created `tClass` will be added to the argument passed in, in the latter case it will be added to the parent of the argument passed in, which seems confusing IMHO. Why not always adding the new `tClass` to `target`, i.e. always passing in the element that will be the container of the newly created `tClass`?

Reply:
What you describe is true for three instances (Class, Function, ObjectLiteral). What you propose means either introduce another parameter or change the target parameter. Of course your proposal still can be done, but this would break the systematically structured calls in `N4JSTypesBuilder` in the bodies of the _dispatch method_ named `createType`. In case you prefer also to pass the new argument through right from `buildTypes` it should be noted that also all methods of `relinkType` were affected since they currently are implemented similarly.

Reply:
Ok, sorry. My misunderstanding. I confused `#internalTypes` with `#topLevelTypes`.